gammapy/region/circle.py

```diff
@@ -0,0 +1,170 @@
+from astropy.wcs.utils import skycoord_to_pixel, pixel_to_skycoord
+from photutils.utils.wcs_helpers import (
+    skycoord_to_pixel_scale_angle,
+)
+from .core import SkyRegion, PixRegion
+
+
+__all__ = [
+    'PixCircleRegion',
+    'SkyCircleRegion',
+]
+
+
+
+class PixCircleRegion(PixRegion):
+    """
+    Circular aperture(s), defined in pixel coordinates.
+    Parameters
+    ----------
+    pos : tuple, list, array
+        Pixel coordinates of the circle center
+    radius : float
+        Circle radius, in pixels
+    """
+
+    def __init__(self, pos, radius):
+
+        self.pos = (pos[0], pos[1])
+        self.radius = radius
+
+    def to_sky(self, wcs, frame='galactic'):
+        """
+        Return a `~gammapy.regions.SkyCircleRegion`.
+
+        Parameters
+        ----------
+        wcs : `~astropy.wcs.WCS`
+            WCS object
+        """
+        val = pixel_to_skycoord(self.pos[0], self.pos[1], wcs, mode='wcs', origin=1)
+        if frame == 'galactic':
+            sky_position = val.galactic
+        elif frame == 'icrs':
+            sky_position = val.icrs
+        
+        xc, yc, scale, angle = skycoord_to_pixel_scale_angle(sky_position, wcs)
+        val = (self.radius * u.pix / scale).to(u.deg)
+        sky_radius = np.round(val,4)
+        return SkyCircleRegion(sky_position, sky_radius)        
+
+    def offset(self, pos):
+        """
+        Compute offset wrt to a certain pixel position
+
+        Parameters
+        ----------
+        pos : tuple
+            Position to which offset is computed
+    
+        Returns
+        -------
+        offset : float
+            Offset in pix
+
+        """
+        x2 = (self.pos[0] - pos[0]) ** 2
+        y2 = (self.pos[1] - pos[1]) ** 2
+        offset = np.sqrt(x2 + y2) 
+        return offset
+
+    def angle(self, pos):
+        """
+        Compute angle [rad] wrt to a certain pixel position
+
+        Parameters
+        ----------
+        pos : tuple
+            Position to which angle is computed
+
+        Returns
+        -------
+        angle : float
+            Angle in rad
+        """
+        dx = self.pos[0] - pos[0]
+        dy = self.pos[1] - pos[1]
+        angle = np.arctan2(dx, dy)
+        return angle
+
+    def is_inside_exclusion(self, exclusion_mask):
+        """
+        Check if region overlaps with a given exclusion mask
+
+        Parameters
+        ----------
+        exclusion_mask : `~gammapy.region.ExclusionMask`
+            Exclusion mask
+
+        Returns
+        -------
+        bool
+        """
+        from ..image import lookup
+        x,y = self.pos
+        excl_dist = exclusion_mask.distance_image
+        val = lookup(excl_dist, x, y, world=False)
+        return val < self.radius
+
+    def to_mpl_artist(self, **kwargs):
+        """Convert to mpl patch.
+        """
+        xc, yc = self.pos
+        major = self.radius
+        patch = mpatches.Ellipse((xc, yc), 2*major, 2*major, angle=0, **kwargs)
```

Comment:
> Shouldn't the MPL patch reflect the actual region and not a scaled version?

Yes.

(I thought the factor 2 was a scaled version to make it bigger, but now I see that `Ellipse` takes diameter as parameter.)

Why not use [Circle](http://matplotlib.org/api/patches_api.html#matplotlib.patches.Circle) from matplotlib to represent a circle? :smile: